### DISCUSSION TITLE
redirect after authentication to wrong port

### DISCUSSION THREAD

**Discussion by MohamedAnouar** [OpenIDC/mod_auth_openidc#868] [Q&A] [answered]
Hello,

I had my configuration working perfectly on apache 2.4.46, but once I updated the config to 2.4.53, I noticed an issue that is linked to the last redirection after code validation. https://example.com/?code... : code validation is OK.
The user got redirected to https://example.com:8080/app instead of [https://example:443/app](https://example/app) the external route (exposed via openshift). OIDC Module is forcing the use of virtual host port instead of the ServerName.
Normally we expect after all the authentication is good that the user should be redirected to /app without any dns. I tried modifying the location header for the last redirection but without success. Do you have a fix?

**Comment by zandbelt** [comment, marked as answer]
you'll need to make the module aware of the external port with a header that is set by the LB/proxy running in front of it, i.e. it should add the header `X-Forwarded-Port 443` (and you'll need to add `OIDCXForwardedHeaders X-Forwarded-Port ` to the module's config if you're running a recent version of it), see: https://github.com/zmartzone/mod_auth_openidc/wiki#8-how-do-i-run-mod_auth_openidc-behind-a-reverse-proxy

**Reply by MohamedAnouar** [reply]
Perfect !!! Thank a lot for you help I was struggling with this issue. Resolved by using OIDCXForwardedHeaders X-Forwarded-Host
Thanks again
